Packs/Cognni/Integrations/Cognni/Cognni.py

```diff
@@ -0,0 +1,690 @@
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='Cognni.insight',
+        outputs_key_field='id',
+        outputs=insight
+    )
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    api_key = demisto.params().get('apikey')
+
+    # get the service API url
+    base_url = demisto.params()['url']
+
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    first_fetch_time = arg_to_timestamp(
+        arg=demisto.params().get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True
+    )
+    assert isinstance(first_fetch_time, int)
+
+    proxy = demisto.params().get('proxy', False)
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy)
+
+        fetch_key_res = client.fetch_key(api_key)
+        access_token = fetch_key_res['token']
+
+        headers = {
+            'Authorization': f'Bearer {access_token}',
+            'Content-Type': 'application/json',
+        }
+
+        client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy)
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            return_results(result)
+
+        elif demisto.command() == 'cognni-ping':
+            return_results(ping_command(client))
+
+        elif demisto.command() == 'cognni-fetch-incidents':
+            return_results(fetch_incidents_command(client, first_fetch_time, demisto.args()))
+
+        elif demisto.command() == 'fetch-incidents':
+            min_severity = demisto.params().get('min_severity', None)
+
+            max_fetch = arg_to_int(
+                arg=demisto.params().get('max_fetch'),
+                arg_name='max_fetch',
+                required=False
+            )
+            if not max_fetch or max_fetch > MAX_EVENTS_TO_FETCH:
+                max_fetch = MAX_EVENTS_TO_FETCH
+
+            next_run, incidents = fetch_incidents(
+                client=client,
+                events_limit=max_fetch,
+                last_run=demisto.getLastRun(),  # getLastRun() gets the last run dict
+                first_fetch_time=first_fetch_time,
+                min_severity=min_severity
+            )
+
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+
+        elif demisto.command() == 'cognni-get-event':
+            return_results(get_event_command(client, demisto.args()))
+
+        elif demisto.command() == 'cognni-fetch-insights':
+            return_results(fetch_insights_command(client, demisto.args()))
```

Comment:
Could this command fetch a *lot* of insights? If so I recommend adding some sort of limit to the number of insights it could fetch at a time as an argument and set a max limit.